dds/DCPS/RTPS/Sedp.cpp

```diff
@@ -4013,36 +4014,46 @@ Sedp::add_security_info(const DCPS::TransportLocatorSeq& locators,
                         const RepoId& writer, const RepoId& reader)
 {
   using DCPS::Serializer;
+  using namespace DDS::Security;
 
   if (std::memcmp(writer.guidPrefix, reader.guidPrefix,
                   sizeof(DCPS::GuidPrefix_t)) == 0) {
     // writer and reader belong to the same participant, no security needed
     return locators;
   }
 
-  DDS::Security::ParticipantCryptoHandle part_handle = DDS::HANDLE_NIL;
-  DDS::Security::DatawriterCryptoHandle dw_handle = DDS::HANDLE_NIL;
-  DDS::Security::DatareaderCryptoHandle dr_handle = DDS::HANDLE_NIL;
+  DCPS::RepoId remote_part;
+  const DCPS::RepoId local_part = spdp_.guid();
+  if (std::memcmp(writer.guidPrefix, local_part.guidPrefix, sizeof writer.guidPrefix) == 0) {
+    remote_part = reader;
+  } else if (std::memcmp(reader.guidPrefix, local_part.guidPrefix, sizeof reader.guidPrefix) == 0) {
+    remote_part = writer;
+  } else {
+    return locators;
```

Comment:
That should be caught much earlier.  By the time it gets here we are handling an association which has to have at least one side local.  If both sides are local it should still work since the `lookup_participant_crypto_info` a few lines below this will return nil.